src/main/main.ts

```diff
@@ -572,16 +572,16 @@ const doSplashScreenChecks = async () =>
             await sleep(250);
 
             splash.webContents.send('checking-python');
-            await checkPythonEnv(splash);
+            const pythonInfo = await checkPythonEnv(splash);
             await sleep(250);
 
             splash.webContents.send('checking-deps');
             const hasNvidia = await nvidiaSmiPromise;
-            await checkPythonDeps(splash, hasNvidia);
+            await checkPythonDeps(splash, pythonInfo, hasNvidia);
             await sleep(250);
 
             splash.webContents.send('spawning-backend');
-            await spawnBackend(port);
+            spawnBackend(port, pythonInfo);
```

Comment:
One of the improvements is here. We can now clearly see the data dependency between `checkPythonEnv` and the below functions. We have to know which python to use before we can install dependencies and spawn the backend. Clearly expressed in code. Global state used to hide this from us.